pkg/cmd/cli/cmd/env.go

```diff
@@ -324,24 +333,28 @@ func RunEnv(f *clientcmd.Factory, in io.Reader, out io.Writer, cmd *cobra.Comman
 
 	failed := false
 	for i, info := range infos {
-		newData, err := json.Marshal(objects[i])
-		if err != nil {
-			return err
-		}
-		patchBytes, err := strategicpatch.CreateTwoWayMergePatch(oldData[i], newData, objects[i])
-		if err != nil {
-			return err
-		}
-		obj, err := resource.NewHelper(info.Client, info.Mapping).Patch(info.Namespace, info.Name, kapi.StrategicMergePatchType, patchBytes)
-		if err != nil {
-			handlePodUpdateError(cmd.Out(), err, "environment variables")
-			failed = true
-			continue
+		outputMsg := "updated"
+		if !updated[i] {
+			outputMsg = "nothing to update"
```

Comment:
Also it would be nice if this (no need for update) was handled by `CreateTwoWayMergePatch` by returning an empty byte slice and not by us but that's not the case: https://github.com/kubernetes/kubernetes/issues/17022